PulseEco/Managers/LocationManager.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  LocationManager.swift
+//  PulseEco
+//
+//  Created by Nikola Jankovikj on 11.11.24.
+//
+
+import Foundation
+import CoreLocation
+
+class LocationManager: NSObject, ObservableObject {
+    private let manager = CLLocationManager()
+    static let shared = LocationManager()
+    
+    @Published var currentUserLocation: CLLocationCoordinate2D?
+    @Published var currentUserCity: String?
+    
+    override init() {
+        super.init()
+        manager.delegate = self
+        manager.startUpdatingLocation()
+    }
+    
+    func requestLocation() {
+        manager.requestWhenInUseAuthorization()
+    }
+    
+    func isAuthorizedOrNotDetermined() -> Bool {
```

Comment:
we want to have custom AuthorisationStatus which will return status(authorised/not authorised/ not determened) and not bool